sdk/python/kfp/cli/component.py

```diff
@@ -214,6 +215,13 @@ def _load_components(self):
             raise sys.exit(1)
         logging.info(f'Using target image: {self._target_image}')
 
+        pip_index_urls = []
+        for comp in self._components:
+            if comp.pip_index_urls is not None:
+                pip_index_urls.extend(comp.pip_index_urls)
+        if len(pip_index_urls) > 0:
+            self._pip_index_urls = list(dict.fromkeys(pip_index_urls))
```

Comment:
Can you explain what's happening here? It seems like this converts a list to dict then back to list, but it's possible I'm missing something.

Reply:
It is a little trick to get rid of possible duplicates preserving order (dict guarantees insertion order since 3.7, see [here](https://docs.python.org/3/library/stdtypes.html?highlight=dict#dict.popitem).
I will remove this as it is a bit too defensive and it is not done for any other list arguments I've seen - to be consistent.

Reply:
Neat! Thank you for explaining and good thinking.

I actually think it makes sense to keep this deduping. I can imagine containerizing many components that each use the same single URL. We don't need to list that package registry multiple times. This is probably the most realistic use case, since combining the `pip_index_urls` of components that each use a different registry wouldn't really make sense.

And the order preserving is good, too. There's no logical total ordering across multiple components, but it's easy enough to preserve a partial ordering, so let's do it.

The entries in the requirements file (`runtime-requirements.txt`) are deduplicated (via `set()`) and have some of the same concerns/considerations that arise from combining the component parameters for multiple components, so there's one precedent for deduping.

WDYT?

Reply:
Neat! Thank you for explaining and good thinking.

I actually think it makes sense to keep this deduping. I can imagine containerizing many components that each use the same single URL. We don't need to list that package registry multiple times. This is probably the most realistic use case, since combining the `pip_index_urls` of components that each use a different registry wouldn't really make sense.

And the order preserving is good, too. There's no logical total ordering across multiple components, but it's easy enough to preserve a partial ordering, so let's do it.

The entries in the requirements file (`runtime-requirements.txt`) are deduplicated (via `set()`) and have some of the same concerns/considerations that arise from combining the component parameters for multiple components, so there's one precedent for deduping.

WDYT?

Reply:
You're right, the deduplication is actually necessary when more components in a file share the same pypi url, dockerfile needs it only once - what is I guess expected use-case - company having own pypi index.
I've put it back in :)
Preserving order is nice to have for unit tests, otherwise set would be fine (like in runtime-requirements case).